packages/x-components/src/x-modules/search/store/actions/fetch-and-save-search-response.action.ts

```diff
@@ -7,8 +7,15 @@ const { fetchAndSave, cancelPrevious } = createFetchAndSaveActions<
   SearchRequest | null,
   SearchResponse
 >({
-  fetch({ dispatch }, request) {
-    return dispatch('fetchSearchResponse', request);
+  fetch({ dispatch, state }, request) {
+    const newRequest = request
+      ? {
+          ...request,
+          // eslint-disable-next-line @typescript-eslint/no-extra-parens
+          ...(state.origin && { origin: state.origin })
+        }
+      : null;
+    return dispatch('fetchSearchResponse', newRequest);
```

Comment:
Is it not easier this?
```suggestion
  fetch({ dispatch, state: { origin } }, request) {
    if (request && origin) {
      request.origin = origin;
    }
    return dispatch('fetchSearchResponse', request);
  },
```